### PR TITLE
[Coverity Scan Issue] fix cs issues in scaled_attn.cpp

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Fix the Coverity Scan Issue in CPU Plugin scaled_attn.cpp

### Tickets:
 - 125936

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

File: src/plugins/intel_cpu/src/nodes/scaled_attn.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ struct MHAKernel {
     }
 
     PlainTensor<uint8_t> causal_mask;
-    bool select_nfltmax_at_0;  // set attn_score to -FLT_MAX when causal_mask[...] equal to this
+    bool select_nfltmax_at_0 = false;  // set attn_score to -FLT_MAX when causal_mask[...] equal to this
```
Comment (praasz) [inline src/plugins/intel_cpu/src/nodes/scaled_attn.cpp:80]: Is Coverity reporting this line?
this class has `MHAKernel() = default;` all members if got default ctor will be initialized.

File: src/plugins/intel_cpu/src/nodes/scaled_attn.cpp [hunk 2]
```diff
@@ -452,7 +454,7 @@ struct MHASingleToken {
     MHASingleToken() : m_attn_w(true), m_temp(true) {}
```
Comment (praasz) [inline src/plugins/intel_cpu/src/nodes/scaled_attn.cpp:454]: All members should be initialized here to have correct default initialization

File: src/plugins/intel_cpu/src/nodes/scaled_attn.cpp [hunk 3]
```diff
@@ -303,7 +303,7 @@ struct MHAKernel<ScaledDotProductAttention::KT_MLAS, float> {
     }
 
     PlainTensor<uint8_t> causal_mask;
-    bool select_nfltmax_at_0;  // set attn_score to -FLT_MAX when causal_mask[...] equal to this
+    bool select_nfltmax_at_0 = false;  // set attn_score to -FLT_MAX when causal_mask[...] equal to this
```
Comment (praasz) [inline src/plugins/intel_cpu/src/nodes/scaled_attn.cpp:306]: provide default initialization in `MHAKernel()` ctor.

File: src/plugins/intel_cpu/src/nodes/scaled_attn.cpp [hunk 4]
```diff
@@ -342,6 +342,8 @@ struct MHAKernel<ScaledDotProductAttention::KT_MLAS, float> {
 
         parallel_for3d(B, H, m_blocks, [&](size_t b, size_t h, size_t m_blk) {
             size_t thread_id = static_cast<size_t>(parallel_get_thread_num());
+            if (thread_id < 0)
```
Comment (praasz) [inline src/plugins/intel_cpu/src/nodes/scaled_attn.cpp:345]: Get `parallel_get_thread_num` as signed check if negtive (throw) and then cast?
How `size_t` can be negative?

Comment (zhaixuejun1993) [reply src/plugins/intel_cpu/src/nodes/scaled_attn.cpp:346]: @praasz Try "static_cast<size_t>(-2)" with GCC13.1, get 18446744073709551614.
So I think we don't need to convert to size_t, just need to if negtive (throw) and then use it.

**Review by praasz — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Pull request merged**